src/v/cloud_storage/inventory/utils.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "container/fragmented_vector.h"
+#include "model/fundamental.h"
+
+#include <seastar/core/file.hh>
+#include <seastar/core/iostream.hh>
+
+namespace cloud_storage::inventory {
+
+// Writes hashes for a single NTP to a data file. The data file is
+// located in path: namespace/topic/partition_id/{seq}. The parent
+// directories are created if missing. A new data file is created for each
+// flush operation.
+ss::future<> flush_ntp_hashes(
+  std::filesystem::path root,
+  model::ntp ntp,
+  fragmented_vector<uint64_t> hashes,
+  uint64_t file_name);
+
+ss::future<>
+write_hashes_to_file(ss::file& f, fragmented_vector<uint64_t> hashes);
+
+ss::future<> write_hashes_to_file(
+  ss::output_stream<char>& stream, fragmented_vector<uint64_t> hashes);
+
+} // namespace cloud_storage::inventory
```

Comment:
nit: these all seem to be in service of the single task of writing hashes. Would it make sense to wrap them in some `hash_writer` class that wraps a ss::file?

Or are these separate because they're used individually in tests?

Reply:
Or maybe the write_hashes_to_file() can just be in an anonymous namespace in the cc file?

Reply:
IIRC these were member functions of the inventory consumer which were separate just for readability. When I moved them to independent compilation unit, they are no longer required to be public other than the flush function. I'll make the other two private.